### PR TITLE
flexibleServerPrivateDnsZone

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Private DNS Zone Integrated',
+    category: 'PostgreSQL Server',
+    domain: 'Databases',
+    description: 'Ensure that PostgreSQL flexible servers has Private DNS Zone integrated.',
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.js:8]:
```suggestion
    description: 'Ensure that PostgreSQL flexible servers have private DNS zone integrated.',
```

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    more_info: 'Integrate Private DNS Zones with PostgreSQL flexible servers to enhance DNS service reliability and security within your Azure virtual network, ensuring seamless DNS resolution and streamlined domain management.',
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.js:9]:
```suggestion
    more_info: 'Integrating Private DNS Zones with PostgreSQL flexible servers enhances DNS service reliability and security within your Azure virtual network, ensuring seamless DNS resolution and streamlined domain management.',
```

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    recommended_action: 'Ensures Vnet and Private DNS Zone (private access) is integrated for PostgreSQL flexible server.',
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.js:10]:
```suggestion
    recommended_action: 'Ensure Vnet and Private DNS Zone (private access) is integrated for PostgreSQL flexible server.',
```

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No existing PostgreSQL flexible servers found', location);
+                return rcb();
+            }
+
+            for (let flexibleServer of servers.data) {
+                if (!flexibleServer.id) continue;
+                
+                if (flexibleServer.network && flexibleServer.network.privateDnsZoneArmResourceId) {
+                    helpers.addResult(results, 0, 'PostgreSQL flexible server has Private DNS Zone integrated', location, flexibleServer.id);
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.js:40]:
```suggestion
                    helpers.addResult(results, 0, 'PostgreSQL flexible server has private DNS zone integrated', location, flexibleServer.id);
```

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'PostgreSQL flexible server does not have Private DNS Zone integrated', location, flexibleServer.id);
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.js:42]:
```suggestion
                    helpers.addResult(results, 2, 'PostgreSQL flexible server does not have private DNS zone integrated', location, flexibleServer.id);
```

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.spec.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    "id": "/subscriptions/12345/resourceGroups/Default/providers/Microsoft.DBforPostgreSQL/flexibleServers/test-server2",
+    "type": "Microsoft.DBforPostgreSQL/flexibleServers",
+    "storageProfile": {
+        "storageMB": 5120,
+        "backupRetentionDays": 7,
+        "geoRedundantBackup": "Disabled",
+        "storageAutogrow": "Disabled"
+    },
+    "network":{
+        "delegatedSubnetResourceId" : "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/cloudsploit-dev/providers/Microsoft.Network/virtualNetworks/omer-virtual-network-test/subnets/default",
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.spec.js:28]:
```suggestion
        "delegatedSubnetResourceId" : "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/cloudsploit-dev/providers/Microsoft.Network/virtualNetworks/omer-virtual-network-test/subnets/default",
```
change subscription id to a random value

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/azure/postgresqlserver/flexibleServerPrivateDns.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Private DNS Zone Integrated',
+    category: 'PostgreSQL Server',
+    domain: 'Databases',
+    description: 'Ensure that PostgreSQL flexible servers has Private DNS Zone integrated.',
+    more_info: 'Integrate Private DNS Zones with PostgreSQL flexible servers to enhance DNS service reliability and security within your Azure virtual network, ensuring seamless DNS resolution and streamlined domain management.',
+    recommended_action: 'Ensures Vnet and Private DNS Zone (private access) is integrated for PostgreSQL flexible server.',
+    link: 'https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-networking-private',
```
Comment (mehakseedat63) [inline plugins/azure/postgresqlserver/flexibleServerPrivateDns.js:11]:
```suggestion
    link: 'https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-networking-private',
```
this is the same as vnet integration, can we add a link that mentions private dns zones?

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by alphadev4 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**